docs/installing.rst

```diff
@@ -208,7 +208,185 @@ Congratulations! You have just installed PySyft on Mac OSX!
+version of python but fair warning... e're be dragons.
+
+Step 3 - Setup Environment
+--------------------------
+
+I'm sure you're tempted to skip this step. Word from the wise...
+`Don't skip this step.
+<https://twitter.com/iamtrask/status/1300854373296332809>`_
+
+You are about to install a library with lots of complex dependencies. You don't want to break
+something on your computer because you're installing PySyft. And vice versa, you don't want
+to later break your PySyft install when installing some other tool later! Friends don't
+let friends build libraries from source without using a virtual environment.
+
+And since PySyft uses (and will use more and more) non-python dependencies, the best
+virtual environment to use for PySyft is conda. Note, if you're tempted to use virtualenv
+instead, `read this warning <https://twitter.com/shreyshahi/status/1300855906742140928>`_.
+
+Step 3.1 - Install Conda
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+First, let's see if you have conda installed! Type "conda" into your Terminal app and hit enter.
+
+.. code:: console
+
+    > conda
+    usage: conda [-h] [-V] command ...
+
+    conda is a tool for managing and deploying applications, environments and packages.
+
+    Options:
+
+    positional arguments:
+      command
+        clean        Remove unused packages and caches.
+        config       Modify configuration values in .condarc. This is modeled
+                     after the git config command. Writes to the user .condarc
+                     file (/Users/atrask/.condarc) by default.
+        create       Create a new conda environment from a list of specified
+
+If calling "conda" doesn't return something like this, then you need to install conda. Just
+follow the `installation instructions <https://docs.conda.io/projects/conda/en/latest/user-guide/install/>`_
+and you'll be fine.
+
+Step 3.2 - Create conda Env
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now, we want to create a conda virtual environment with the latest verison of Python supported
+by syft which, at the time of writing, is 3.8.
+
+.. code:: console
+
+    conda create -n my_syft_env python=3.8
+
+Then follow the instructions it gives you to create your environment.
+
+Step 3.3 - Activate Conda Env
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To activate the environment you created in the last step, simply enter `activate <environment name>`
+which if you simply copy pasted the line above, the environment name was "my_syft_env".
+
+.. code:: console
+
+    conda activate my_syft_env
+
+.. note::
+
+    You will need to activate this my_syft_env environment whenever you want to use syft,
+    unless of course you install syft in another environment.
+    
+Step 4 - Install Dependencies
+-----------------------------
+
+Assuming you're still in the base directory of PySyft (as you came to be in step 1), you can
+now install the dependencies that PySyft relies on using the following command:
+
+.. code:: console
+
+    pip install -r requirements.txt
```

Comment:
Q: Shouldn't we use ```setup install -e .```?

Reply:
I used requirements.txt to install dependencies so gave that as tutorial.... if `setup install -e .` works and is better I can change that. What do you suggest?